File: mobile_deckz/lib/model/gamehistory/game.dart
import 'dart:core';

class Game {

  final DateTime timestamp;
  final String deckName;
  final String score;

  Game({required this.timestamp, required this.deckName, required this.score});

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      timestamp: json['timestamp'],
      deckName: json['deckName'].toString(),
      score: json['score'].toString(),
    );
  }

  String getFormattedTimestamp() {
    return timestamp.toString();
  }

  int getScore() {
    return int.parse(score);
  }

  int compareTo(Game o) {
    return o.timestamp.compareTo(timestamp);
  }
}

File: src/ulb/infof307/g01/model/gamehistory/Game.java
package ulb.infof307.g01.model.gamehistory;

import java.text.SimpleDateFormat;

public record Game(long timestamp, String deckName, String score) implements Comparable<Game> {

    public String getFormattedTimestamp() {
        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
        return sdf.format(timestamp);
    }

    public int getScore() {
        return Integer.parseInt(score);
    }

    @Override
    public int compareTo(Game o) {
        return Long.compare(o.timestamp, timestamp);
    }
}
